doc/src/pair_lepton.rst

```diff
@@ -0,0 +1,125 @@
+.. index:: pair_style lepton
+.. index:: pair_style lepton/omp
+
+pair_style lepton command
+=========================
+
+Accelerator Variants: *lepton/omp*
+
+Syntax
+""""""
+
+.. code-block:: LAMMPS
+
+   pair_style style args
+
+* style = *lepton*
+* args = list of arguments for a particular style
+
+.. parsed-literal::
+
+    *lepton* args = cutoff
+      cutoff = global cutoff for the interactions (distance units)
+
+Examples
+""""""""
+
+.. code-block:: LAMMPS
+
+   pair_style lepton 2.5
+
+   pair_coeff  * *  "k*((r-r0)^2*step(r0-r)); k=200; r0=1.5" 2.0
+   pair_coeff  1 2  "4.0*eps*((sig/r)^12 - (sig/r)^6);eps=1.0;sig=1.0" 1.12246204830937
+   pair_coeff  2 2  "eps*(2.0*(sig/r)^9 - 3.0*(sig/r)^6);eps=1.0;sig=1.0"
+
+Description
+"""""""""""
+
+.. versionadded:: TBD
+
+Pair style *lepton* computes spherical pairwise interactions based on
```

Comment:
replace "computes spherical pairwise ..." with something like

computes pairwise interactions between particles which depend solely on the distance
between the particles, for pairs of particles within the specified cutoff.  The potential function
(energy) must be specified as an expression ...

Am I correct this style cannot be used (for example) with potentials that depend on charge?

Reply:
> Am I correct this style cannot be used (for example) with potentials that depend on charge?

This specific style not, but I can imagine doing a variant that does. It is possible to have multiple "external variables" in a Lepton expression that can be set on a per-evaluation basis. But that comes with significant overhead for cases that do not use it. I have been thinking about interfacing other properties. For equal style variables, I have decided on doing a string replacement instead because that only needs to be done once and then the expression compiler can do more optimizations. Interfacing charges would be somewhat easier, but I can also imagine an interface to per-atom custom properties or atom-style variables. This mostly requires some conventions and possibly writing some extra C++ classes for the Lepton library implementing functions that access the data.

Reply:
> Am I correct this style cannot be used (for example) with potentials that depend on charge?

Support for implementing coulomb interactions has been added as a new pair style `lepton/coul`. For this pair style you can reference either "qi" or "qj" within the expression to get access to the charge. More details and the differences between this style and the plain `lepton` one are in the docs. As you may see from the code, while Lepton has convenient ways to set values in such expressions, they come with some overhead and because the expression is evaluated for each pair, there is only a limited amount of the processing that can be done outside the innerloop. Trying to access per-atom variables or properties would be even more complex (global variables are simple in comparison and can be done with a string replacement just as like in input files and `${variable}` expansions).

Reply:
@sjplimp FYI compatibility with kspace styles is implemented the same way as for pair style table. I have been able to create tables with this pair style that agree with those for coul/cut and coul/long within 1.0e-14 in forces and energies.